server/axios_api.js

```diff
@@ -0,0 +1,62 @@
+var axios = require("axios").default;
+var HOST = "movies-tvshows-data-imdb.p.rapidapi.com";
+var API_KEY = "030611f594msh73349b7b00af6e8p16d57ejsncbf1b45a6165";
+var URL = "https://movies-tvshows-data-imdb.p.rapidapi.com/";
+
+
+
+function get_details(id) {
+  var options = {
+    method: 'GET',
+    url: URL,
+    params: {type: 'get-movie-details', imdb: id},
+    headers: {
+      'x-rapidapi-host': HOST,
+      'x-rapidapi-key': API_KEY
+    }
+  };
+  axios.request(options).then(function (response) {
```

Comment:
can we use async await here?

Reply:
@isha-bansal0115 I think so, I'm just not very familiar with how to properly write javascript functions. Where would I put the async-await calls?

Reply:
yeah sure. so we can do this,

const get_details = async(id) => {

try {
const apiResponse = await axios.request(options);
<process response as you have done above>
} catch(error) => {}


}

This is much cleaner than traditional .then promises. I do see you have used it below during get_id

Reply:
Do we need async method? There is no point of having an async here, I assume. Request and fetch would do for our use-case.
@isha-bansal0115 Let me know your thoughts

Reply:
async is a modern way to write asynchronous code handling. If we use async we can avoid .then.

fetch is async call so either we do .then or we use await. Using async await is modern and concise.

Reply:
@isha-bansal0115 Are you suggesting it should look like this?
`const get_details = async(id)=> {
  var options = {
    method: 'GET',
    url: URL,
    params: {type: 'get-movie-details', imdb: id},
    headers: {
      'x-rapidapi-host': HOST,
      'x-rapidapi-key': API_KEY
    }
  };
  try{
    const response = axios.request(options);
    console.log(response)
    console.log("rating:"+response.data["imdb_rating"]);    
  }
  catch(error){
    console.error(error);
  }
}`

Reply:
yes sir. if you define this as async function then u can use await syntax inside and since axios.request is async call then we can wait for it to resolve. async await is syntactic sugar so its more concise.